### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by Luckyaremu**
#Enumerables methods

- Rspec testing methods done from my_each to my_multiply
-All linters requirements properly done
-The linter error showing is as a result of bug from linters

### CODE REVIEW

**Comment by euqueme** [comment]
Hello @Luckyaremu I can't start the review because I don't see your coding partner commits or as a participant at all but the code review request was made as a team,

:pray: Please if both of you worked evenly in this project state it in your Pull Request description and add Kotoku Kennedy as a collaborator.

:pray: If you worked alone in this project please mark the checkbox that says you worked alone in this project when you submit for a code review again

PROJECT STATUS: INVALID :stop_sign:

**Comment by kotoku-code** [comment]
RSPEC for enumerable project
- Used Rspec
- Fixed linter error
- Updated Readme

Thank you for reviewing our code, kindly let us know if there are any corrections needed to be made.

kind regards

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  end
+
+  describe '#my_all' do
+    it 'returns true if all elements matches' do
+      expect(%w[ant bear cat].my_all? { |word| word.length >= 3 }).to eql(true)
+    end
+    it 'returns true if all the elements match the parameter' do
+      expect([3, 4, 7, 11].my_all?(3)).to eql(false)
+    end
+  end
```
Comment (mosesogwo) [inline spec/enumerable_spec.rb]: When does it return false? Write a test for it.

File: spec/enumerable_spec.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  describe '#my_any' do
+    it 'returns true if all the elements matches' do
+      expect(%w[ant bear cat].my_any? { |word| word.length >= 3 }).to eql(true)
+    end
+    it 'returns false if the array is empty' do
+      expect([].my_any?).to eql(false)
+    end
+  end
```
Comment (mosesogwo) [inline spec/enumerable_spec.rb]: What does it return when only some of the elements return true? Write a test for it.
Also write a test for when all the elements return false for the block.

File: spec/enumerable_spec.rb [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+  describe '#my_none' do
+    it 'returns true if all the elements matches' do
```
Comment (mosesogwo) [inline spec/enumerable_spec.rb]: It returns false (not true), if all the elements satisfy the block.

File: spec/enumerable_spec.rb [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      expect(%w[ant bear cat].my_none? { |word| word.length >= 3 }).to eql(false)
+    end
+    it 'returns true if regexp matches all the elements' do
+      expect(%w[ant bear cat].my_none?(/d/)).to eql(true)
+    end
+  end
```
Comment (mosesogwo) [inline spec/enumerable_spec.rb]: When does my_none? return false. Add tests for it.

File: spec/enumerable_spec.rb [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+      expect([1, 2, 4, 2].my_count(2)).to eql(2)
+    end
+    it 'returns the number of elements that matches the block' do
+      expect([1, 2, 4, 2].my_count(&:even?)).to eql(3)
+    end
+  end
+  describe '#multiply_els' do
+    it 'multiplies the elements between each other' do
+    end
+  end
```
Comment (mosesogwo) [inline spec/enumerable_spec.rb]: Where is the test?

File: spec/enumerable_spec.rb [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  describe '#my_map' do
+    it 'returns a new array with the classes of each element of the array' do
+      expect(%w[a b c].my_map(&:class)).to eql([String, String, String])
+    end
+  end
```
Comment (mosesogwo) [inline spec/enumerable_spec.rb]: Write test for when no block is given to my_map

File: spec/enumerable_spec.rb [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  describe '#my_inject' do
+    it 'returns a new array with the classes of each element of the array' do
```
Comment (mosesogwo) [inline spec/enumerable_spec.rb]: my_inject **does not returns a new array with the classes of each element of the array**. Correct the test description.

Add tests for when a symbol is supplied and when an initial value is provided.

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hi @Luckyaremu, @kotoku-code:clap:. Good attempt on testing your methods. Let's make these improvements :fast_forward:**

- Add a space between describe blocks in your spec file.
- For **my_each, my_each_with_index and my_select** specs, add tests for the negative expectations. For example `expect([1, 2, 3, 4, 'hi'].my_each { |x| print x + 1 }).not_to eql([1, 2, 3, 4, 'hi'])`
- For **my_all?, my_any?, my_none?** methods, add tests for the cases where a Class and a Regex is provided as argument.
- Address the comments I made on the spec file
- A better title for your PR will be "RSpec Testing" 
- The test.txt file shouldn't be committed. Your spec file is the appropriate test file.

_**Kindly make these changes and submit another code review request.**_

**Any question? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Comment by Luckyaremu** [comment]
#RSpec Testing

Rspec testing methods updated. More methods added for testing.

**Review by Forison — CHANGES_REQUESTED** [review]
## Staus:: Require changes🛑
# 🙋‍♂️
you have done a good job so far. here are a few areas your test has to cover.
Your tests must cover all the functionalities of each custom method. 
* [ ] ```my_each```, ```my_select```,``` my_each with```, ```my_map``` return enum when no block is given. let us test that functionality.
* [ ]  Some of the arrays used are repeated, let us refactor our tests to have those arrays to become global variables. use ```let(:array_name) { some array }```.
### Happy coding💻
Addo

**Comment by Luckyaremu** [comment]
#RSpec Testing

Reviews treated

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team      
#### Great Job so far, but you haven't implemented changes stipulated by the previous TSE. 
### General
- [ ] Let's implement those changes by previous TSE.
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Comment by Luckyaremu** [comment]
#RSpec Testing

Rspec testing methods updated. More methods added for testing my_each, select, each_with_index.

File: spec/enumerable_spec.rb [hunk 8]
```diff
@@ -24,6 +33,10 @@
     it 'my_select negative case' do
       expect([1, 2, 3, 4, 5, 6].my_select(&:even?)).not_to eql(false)
     end
+    it 'if no block is given return enumerator class' do
+      expect(array.my_select.class).to eq(array.select.class)
+      expect(array.my_select.class).not_to eq('Integer')
+    end
```
Comment (kobiyoyo) [inline spec/enumerable_spec.rb]: You should have one `it` block for each expected behaviour

File: spec/enumerable_spec.rb [hunk 9]
```diff
@@ -109,6 +122,10 @@
     it 'returns an array with the classes of each element' do
       expect(%w[a b c].my_map(&:class)).to eql([String, String, String])
     end
+    it 'if no block is given return enumerator class' do
+      expect(array.my_map.class).to eq(array.map.class)
+      expect(array.my_map.class).not_to eq('Integer')
+    end
```
Comment (kobiyoyo) [inline spec/enumerable_spec.rb]: You should have one it block for each expected behaviour

File: spec/enumerable_spec.rb [hunk 10]
```diff
@@ -16,6 +21,10 @@
     it 'my_each_with_index negative case' do
       expect([1, 2, 3, 4, ''].my_each_with_index.class.equal?([1, 2, 3, 4, ''].select.class)).not_to eq(false)
     end
+    it 'if no block is given return enumerator class' do
+      expect(array.my_each_with_index.class).to eq(array.my_each_with_index.class)
+      expect(array.my_each_with_index.class).not_to eq('Integer')
+    end
```
Comment (kobiyoyo) [inline spec/enumerable_spec.rb]: You should have one it block for each expected behaviour

File: spec/enumerable_spec.rb [hunk 11]
```diff
@@ -1,13 +1,18 @@
...
     it 'returns every single element of an array ' do
       expect([1, 2, 3, 4, 'hi'].my_each { |x| print x }).to eql([1, 2, 3, 4, 'hi'])
     end
     it 'my_each negative case' do
       expect([1, 2, 3, 4, ''].my_each.class.equal?([1, 2, 3, 4].select.class)).not_to eq(false)
     end
+    it 'if no block is given return enumerator class' do
+      expect(array.my_each.class).to eq(array.each.class)
+      expect(array.my_each.class).not_to eq('Integer')
+    end
```
Comment (kobiyoyo) [inline spec/enumerable_spec.rb]: You should have one it block for each expected behaviour

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team      
#### Great Job so far, but there are a few more things required. 
### General
- [ ] Check comments below .


### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Comment by Luckyaremu** [comment]
#RSpec Testing

Rspec testing methods updated. Methods fixed for testing my_each, my_map, select, each_with_index.

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: 

You can merge it :+1: 
Good luck with the next step :handshake: 

Mael FOSSO
